tests/usb_interfaces.test.cpp

```diff
@@ -0,0 +1,737 @@
+      [](scatter_span<hal::byte const> p_data) { std::ignore = p_data; }, 0, 1);
+    std::string expected_iface_name = "Mock";
+    auto res = iface.write_string_descriptor(
+      [&expected_iface_name](scatter_span<hal::byte const> p_data) {
+        std::string_view sv(reinterpret_cast<char const*>(p_data[1].data()),
+                            expected_iface_name.length());
+
+        // Test return string
+        expect(that % expected_iface_name == sv);
+      },
+      1);
+
+    // Test that the method succeeded
+    expect(that % true == res);
+  };
+
+  "usb_interface get_status"_test = [iface]() mutable {
+    std::ignore = iface.write_descriptors(
+      [](scatter_span<hal::byte const> p_data) { std::ignore = p_data; }, 0, 1);
+    req_bitmap iface_bm(
+      true, req_bitmap::type::standard, req_bitmap::recipient::interface);
+
+    auto actual_res_iface = iface.get_status(iface_bm, 0);
+
+    expect(that % 0 == actual_res_iface);
+
+    req_bitmap ep_bm(
+      true, req_bitmap::type::standard, req_bitmap::recipient::endpoint);
+
+    auto actual_res_ep = iface.get_status(ep_bm, 1);
+
+    expect(that % 1 == actual_res_ep);
+  };
+
+  "usb_interface manage alt_settings"_test = [iface]() mutable {
+    iface.set_interface(1, 0);
+
+    auto actual_alt_setting = iface.get_interface(0);
+
+    expect(that % 1 == actual_alt_setting);
+  };
+
+  "usb_interface handle_request"_test = [iface]() mutable {
+    req_bitmap bm = req_bitmap(
+      true, req_bitmap::type::standard, req_bitmap::recipient::interface);
+
+    u8 req = 0x01;
+    u16 value = 0x0203;
+    u16 index = 0x0405;
+    u16 length = 0x0607;
+
+    auto value_bytes = to_le_bytes(value);
+    auto index_bytes = to_le_bytes(index);
+    auto length_bytes = to_le_bytes(length);
+
+    std::array<byte const, 8> expected_payload{
+      bm.m_bitmap,     req,
+      value_bytes[0],  value_bytes[1],
+      index_bytes[0],  index_bytes[1],
+      length_bytes[0], length_bytes[1]
+    };
+    auto actual_res = iface.handle_request(
+      bm,
+      req,
+      value,
+      index,
+      length,
+      [&expected_payload](scatter_span<hal::byte const> p_data) {
+        auto a = make_scatter_bytes(expected_payload);
+        scatter_span<byte const> expected_ss(a);
+
+        expect(that % (expected_ss == p_data));
+      });
+
+    expect(that % true == actual_res);
+  };
+
+  "usb_interface manage_features"_test = [iface]() mutable {
+    req_bitmap ep_bm(
+      true, req_bitmap::type::standard, req_bitmap::recipient::endpoint);
+
+    iface.manage_features(ep_bm, false, 1, 0);
+    expect(that % false == iface.m_generic_ep.info().stalled);
+
+    req_bitmap iface_bm(
+      true, req_bitmap::type::standard, req_bitmap::recipient::interface);
+
+    iface.manage_features(iface_bm, true, 0, 0);
+
+    expect(that % true == iface.m_test_feature);
+
+    iface.manage_features(iface_bm, true, 1, 0);
+
+    expect(that % true == iface.m_another_test_feature);
```

Comment:
Can you verify that all of these expect values were set to their opposites before exercising the APIs? That way we know that we changed something. That would be apart of the "setup" stage of the tests. Replicate the same set of "setup" operations at the top of the test for the rest of them as well. (sorry for the additional work)